model_catalogs/model_catalogs/examples.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+#-*- coding: utf-8 -*-
+"""A module for common components for scripting examples."""
+import time
+from argparse import ArgumentParser
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import List, Tuple
+
+import pandas as pd
+
+import model_catalogs as mc
+
+
+class Timer:
+    """A class to aid with measuring timing."""
+    def __init__(self, msg=None):
+        """Initializes the current time."""
+        self.t0 = time.time()
+        self.msg = msg
+
+    def tick(self):
+        """Update the timer start."""
+        self.t0 = time.time()
+
+    def tock(self) -> float:
+        """Return elapsed time in ms."""
+        return (time.time() - self.t0) * 1000.
+
+    def __enter__(self):
+        """With context."""
+        self.tick()
+
+    def __exit__(self, type, value, traceback):
+        """With exit."""
+        if self.msg is not None:
+            print(self.msg.format(self.tock()))
+
+
+STANDARD_NAMES = [
+    'eastward_sea_water_velocity',
+    'northward_sea_water_velocity',
+    'eastward_wind',
+    'northward_wind',
+    'sea_water_temperature',
+    'sea_water_practical_salinity',
+    'sea_floor_depth',
+]
+
+
+@dataclass
+class FetchConfig:
+    """Configuration data class for fetching."""
+    model_name: str
+    output_pth: Path
+    start: pd.Timestamp
+    end: pd.Timestamp
+    bbox: Tuple[float, float, float, float]
+    timing: str
+    standard_names: List[str] = field(default_factory=lambda : STANDARD_NAMES)
+
+
+def parse_bbox(val: str) -> Tuple[float, float, float, float]:
+    """Return the bounding box parsed from the comma-delimited string.
+
+    Parameters
+    ----------
+    val : str
+        Comma-delimited sequence of 4 float values for (lon_min, lat_min, lon_max, lat_max)
+
+    Returns
+    -------
+        tuple of 4 floats
+    """
+    values = val.split(',')
+    if len(values) != 4:
+        raise ValueError('bbox should include four numbers: lon_min,lat_min,lon_max,lat_max')
+    return tuple(*[float(i) for i in values])
```

Comment:
I think this doesn't work -- tuple() takes a single iterable as an arguement.

`tuple(float(i) for i in values)`

should work, and be the most efficient.

Reply:
yes, good catch!

Reply:
BTW -- there's a bit of bounding box handling code in:

libgoods.utilities

https://github.com/NOAA-ORR-ERD/LibGOODS/blob/main/libgoods/utilities.py

(it's even tested :-) )

maybe not useful, but it would be good to put all of that in one place, and have it be consistent.

Reply:
:+1: